plutus-core/untyped-plutus-core/test/Flat/Spec.hs

```diff
@@ -0,0 +1,43 @@
+-- editorconfig-checker-disable-file
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE FlexibleInstances #-}
+module Flat.Spec (test_flat) where
+
+import Data.Word
+import Flat
+import PlutusCore.DeBruijn
+import Test.Tasty
+import Test.Tasty.QuickCheck
+import UntypedPlutusCore ()
+import UntypedPlutusCore.Core.Type
+
+test_flat :: TestTree
+test_flat = testGroup "FlatProp"
+    [ test_iso
+    , test_deBruijnTripping
+    , test_binderDecode
+    ]
+
+deriving via Word64 instance Arbitrary DeBruijn
+instance Arbitrary FakeNamedDeBruijn where
+    arbitrary= toFake <$> arbitrary -- via debruijn
+deriving newtype instance Arbitrary (Binder DeBruijn)
+deriving newtype instance Arbitrary (Binder FakeNamedDeBruijn)
+
+test_iso :: TestTree
+test_iso = testProperty "fakeIso" $ \d fnd ->
+         (d === fromFake (toFake d))
+    .&&. (fnd === toFake (fromFake fnd))
```

Comment:
These look like two distinct properties, why merge them into a single one? If it ever fails, somebody will need to spend time figuring out which one has failed. Plus you get worse coverage as `d` staying constant and `fnd` changing is a considered a different test case, even though it's not different for `d`.